Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -232,8 +302,32 @@ - (double)doubleForKey:(NSString*)defaultName {
                                                      ([number isKindOfClass:[NSNumber class]] ? [number doubleValue] : 0.0);
 }
 
+- (void) _scheduleSynchronize {
+    // Up to 2 synchronize operations are allowed in the queue, so that if an existing operation has not been removed from the queue,
+    // we still get a synchronize after the CFPreferencesSetAppValue call. Any additional operations would result in an extra synchronize.
+    if ([_synchronizeQueue operationCount] < 2) {
+        [_synchronizeQueue addOperationWithBlock:^void(void) {
+            [self synchronize];
+        }];
+    }
+}
+
+- (NSOperationQueue*) _synchronizeQueue {
+    return _synchronizeQueue;
+}
+
+- (void) _suspendSynchronize {
+    [_synchronizeQueue setSuspended:YES];
+    [_synchronizeQueue waitUntilAllOperationsAreFinished];
+}
```

Comment:
That should not a problem because, _suspendSynchronize is called just before the application is being suspended, so the other thread will also get suspended and when the application resumes _resumeSynchronize will get called on the UI thread and that will resume the other thread from the wait.

---
In reply to: [70900447](https://github.com/Microsoft/WinObjC/pull/612#discussion_r70900447) [](ancestors = 70900447)